src/telemetry/connectionTelemetry.ts

```diff
@@ -38,75 +36,78 @@ type CloudInfo = {
-      };
-    }
+  const cloudInfo: {
+    isAws?: boolean;
+    isAzure?: boolean;
+    isGcp?: boolean;
+  } = await getCloudInfo(firstServerHostname);
 
+  if (cloudInfo.isAws) {
     return {
-      isPublicCloud: false,
-      publicCloudName: null,
+      isPublicCloud: true,
+      publicCloudName: 'aws',
+    };
+  }
+  if (cloudInfo.isGcp) {
+    return {
+      isPublicCloud: true,
+      publicCloudName: 'gcp',
+    };
+  }
+  if (cloudInfo.isAzure) {
+    return {
+      isPublicCloud: true,
+      publicCloudName: 'azure',
     };
-  } catch (error) {
-    log.error('TELEMETRY cloud info error', error);
-
-    return {};
   }
+
+  return {
+    isPublicCloud: false,
+    publicCloudName: null,
+  };
 }
 
 export async function getConnectionTelemetryProperties(
   dataService: DataService,
   connectionType: ConnectionTypes
 ): Promise<NewConnectionTelemetryEventProperties> {
-  const connectionString = dataService.getConnectionString();
-  const firstHost = connectionString.hosts[0] || '';
-  const [hostname] = firstHost.split(':');
-  const authMechanism = connectionString.searchParams.get('authMechanism');
-  const username = connectionString.username ? 'DEFAULT' : 'NONE';
-  const authStrategy = authMechanism ?? username;
-  const instance = await dataService.instance();
-  const cloudInfo = await getCloudInfoFromDataService(hostname);
-  const preparedProperties: NewConnectionTelemetryEventProperties = {
-    auth_strategy: authStrategy,
-    is_atlas: mongoDBBuildInfo.isAtlas(connectionString.toString()),
-    is_localhost: mongoDBBuildInfo.isLocalhost(connectionString.toString()),
-    is_data_lake: instance.dataLake.isDataLake,
-    is_enterprise: instance.build.isEnterprise,
-    is_public_cloud: cloudInfo.isPublicCloud,
-    dl_version: instance.dataLake.version,
-    public_cloud_name: cloudInfo.publicCloudName,
-    is_genuine: instance.genuineMongoDB.isGenuine,
-    non_genuine_server_name: instance.genuineMongoDB.dbType,
-    server_version: instance.build.version,
-    server_arch: instance.host.arch,
-    server_os: instance.host.os,
+  let preparedProperties: NewConnectionTelemetryEventProperties = {
     is_used_connect_screen: connectionType === ConnectionTypes.CONNECTION_FORM,
     is_used_command_palette:
       connectionType === ConnectionTypes.CONNECTION_STRING,
     is_used_saved_connection: connectionType === ConnectionTypes.CONNECTION_ID,
     vscode_mdb_extension_version: version,
   };
 
+  try {
+    const connectionString = dataService.getConnectionString();
+    const firstHost = connectionString.hosts[0] || '';
+    const [hostname] = firstHost.split(':');
+    const authMechanism = connectionString.searchParams.get('authMechanism');
+    const username = connectionString.username ? 'DEFAULT' : 'NONE';
+    const authStrategy = authMechanism ?? username;
+    const instance = await dataService.instance();
+    const cloudInfo = await getCloudInfoFromDataService(hostname);
```

Comment:
I know this is existing code, but there's probably no reason not to run this in parallel, right?

```suggestion
    const [instance, cloudInfo] = await Promise.all([
      dataService.instance(), getCloudInfoFromDataService(hostname)
    ]);
```